### DISCUSSION TITLE
Exporting dataframe with multiple SMILES or Molecule (rdchem.Mol.obj) columns to excel file

### DISCUSSION THREAD

**Discussion by jrkarki** [rdkit/rdkit#6088] [Q&A]
Hello everyone,

I am trying to export my dataframe that contains rdchem.Mol.obj (mol structures) in 3 columns into an .xlsx file. I have followed the instructions from following links and it works fine if I am doing it for just one column.

1 http://rdkit.org/docs/source/rdkit.Chem.PandasTools.html#rdkit.Chem.PandasTools.SaveXlsxFromFrame
2. https://stackoverflow.com/questions/51356324/rdkit-export-pandas-data-frame-with-mol-image 

What I did to write and export one column (Works):

import xlsxwriter
from rdkit.Chem import PandasTools
from rdkit import Chem

df = [sample_data.xlsx](https://github.com/rdkit/rdkit/files/10733893/sample_data.xlsx)

#Code where I convert smiles to Mol and then append it as a new column in df. 
df['MCS Structure'] = [Chem.MolFromSmiles(s) for s in df['MCS']]

#write to file
PandasTools.SaveXlsxFromFrame(df,'/ResultsWithImages.xlsx',molCol='MCS Structure',size=[100,100])

The Problem: Doesn't work for multiple columns

What I did to write 3 columns 

#create mols for each smile columns
df['Frag Structure'] = [Chem.MolFromSmiles(s) for s in df['Frag']]
df['PDB Ligand Structure'] = [Chem.MolFromSmiles(s) for s in df['PDB_ligand']]
df['MCS Structure'] = [Chem.MolFromSmiles(s) for s in df['MCS']]

#Pass all 3 cols to molCol as above
PandasTools.SaveXlsxFromFrame(df,'/Results_Images_AllCols.xlsx',molCol=['Frag Structure','PDB Ligand Structure','MCS Structure'], size = [100,100])

This throws me an error which is => **ValueError: list.remove(x): x not in list**. Also see the screenshot. 
<img width="601" alt="error_rdkit" src="https://user-images.githubusercontent.com/25939327/218780323-55df7257-d64b-4f58-b309-1bce8e91420f.PNG">

What I can think of is to append one column at a time to the final xlsx and overwrite it 2 more times. However, I can't find the option to overwrite the file in SaveXlsxFromFrame. 

Any hints and suggestions?

Thanks in advance.
Reagon

**Comment by greglandrum** [comment]
Hi @jrkarki, sorry for the extremely slow reply.
This is not possible with the current RDKit release, but the functionality has been added to the RDKit already and will be part of the 2023.03 release.

**Reply by jrkarki** [reply]
Thanks Greg. Looking forward for the new release.
